### ISSUE TITLE
Add class for optimizing adaptive KDE rescale factors

### ISSUE THREAD

**Issue by tdent** [tdent/pop-de#32]
For multi-dimensional KDE, in general we want a kernel with different bandwidths along different dimensions.  We also want the overall scale of the kernel to vary between data points (adaptive).  I.e. the kernel for a data point $\vec{X}_i$ is
```math
\exp\left[ -\frac{1}{2} (\vec{x} - \vec{X}_i)^T \lambda_i^{-2} \mathrm{diag}(h_1^{-2}, h_2^{-2}, \ldots) (\vec{x} - \vec{X}_i) \right],
```
where $h_1, h_2, \ldots$ are the global D-dimensional bandwidths and $\lambda_i$ are the per-point adaptive factors. 

The KDEpy implementation (TreeKDE) allows adaptive bandwidths, but enforces that the kernel is proportional to the unit matrix for multi-dimensional data.  So, instead we rescale each dimension (after standardizing it so that the variance is equal to 1). Supposing that $X_i$ are already standardized, we define ${\vec{y} = (h_1^{-1} x_1, h_2^{-1} x_2, \ldots)}^T$ such that the kernel in rescaled coordinates is 
```math
\exp\left[ -\frac{1}{2} (\vec{y} - \vec{Y}_i)^T \lambda_i^{-2} \mathbb{1} (\vec{y} - \vec{Y}_i) \right].
```
This is equivalent to an adaptive KDE with global bandwidth equal to 1.  The D-dimensional bandwidths $h_1,\ldots$, i.e. rescaling factors, have to be optimized (and possibly the adaptive parameter $\alpha$).  As there may be several dimensions this is better done with a standard optimizer such as Nelder-Mead.

**Comment by JAMSADIQ** [comment]
So, one need to write a clas that compute KDE, that is,  the sum of exponentials rather using KDEpy code? 
or we want to pre-compute  ${\vec{y}$ (at which we will compute KDE on)  by already having optimized bandwidths in each dimension?

**Comment by tdent** [comment]
Addressed by #33
